bin/all_sky_search/pycbc_coinc_findtrigs

```diff
@@ -439,4 +468,8 @@ else:
     nslides = 0
 f.attrs['num_slides'] = nslides
 
+for fname in cleanup_files:
```

Comment:
What happens if the job is evicted or something? Does this need to be run in that case?

Reply:
It should. That might depend on how the cluster handles /dev/shm, many delete this when not in use, so might not matter, but I don't know if that's universal. What's the right solution here? Maybe the 'at exit' hook in python?

Reply:
I guess using `with` (and adding the hooks to some object to actually do this cleanup) would ensure this always happens if you get a python error, but an eviction might not allow this to happen.

Still, as you say, /dev/shm should clean itself up, I think.

Reply:
I will add a comment/request: please do not hard-code `/dev/shm` as the location.  The reason for this is that when running under singularity, it is highly desirable to be able to start singularity isolated from the host system (via the `-C` option) in which case the `/dev/shm` inside the container will not be mapped to that outside. One can explicitly mount the host `/dev/shm` to some *other* location, but then that location needs to be configurable, via either command-line argument or environment variable (and it could be reasonable to then default to `/dev/shm` if the configuration option is absent).

Reply:
@josh-willis What is dev/shm mapped to in singularity? The only thing that matters for us here is that it maps to memory as /dev/shm is supposed to. If that's still the case, it doesn't matter if it's not really the same 'filesystem'

Reply:
I think Josh's request makes sense, as this could then also be local disk space (if needed). We (LVK) would have to test the singularity use case as some of our clusters require us to run in this mode.

Reply:
@spxiwh @josh-willis Yeah, having the option to set the 'cache' location make sense anyway, I'll add that option in and rename the options to be a bit more generic.

Reply:
@spxiwh @josh-willis Ok, I've added an option to set the directory and renamed the options to make it more general.

Reply:
@ahnitz Thanks.  To answer your question, when singularity is invoked with `-C`, then it does the following:
```
-C, --containall[=false]      contain not only file systems, but also PID, IPC, and environment
```
In particular, `/dev/shm` will exist in the container, it just won't be mapped to `/dev/shm` on the host.  But you can still expose the host's`/dev/shm` by binding it to an available location inside the container (which of course must exist, and is one reason to make this an option---it depends on where that container and singlarity invocation are looking for it to be mounted).

Note, this way of runing singularity (with `-C`) is the default if HTCondor invokes it, unless the local admin has overridden in (and I also think it's good practice, so that the container is isolated from the host except for things the user has consciously requested).

Reply:
If /dev/shm is as advertised (i.e. a memory mapped directory) then I think in most cases you don't need to use the host's location.

Reply:
i.e. there is not intention to share *between* jobs here, only for the processes spawned by a single one